tensorrt_llm/_torch/auto_deploy/transform/library/sharding.py

```diff
@@ -59,6 +59,24 @@
 )
 
 
+class ShardingTransformConfig(TransformConfig):
+    """Configuration for sharding the model."""
+
+    factory_source: ShardingConfigSource = Field(default=ShardingConfigSource.UNKNOWN)
+    rank: int = Field(default=0)
+    world_size: int = Field(default=1)
+    factory_config: Optional[Dict[str, Any]] = None
+    manual_config: Optional[Dict[str, Any]] = None
+    simple_shard_only: bool = Field(default=False)
+    support_partial_config: bool = False
+    sharding_source: List[ShardingSource] = Field(
+        default_factory=lambda: [ShardingSource.HEURISTIC]
+    )
```

Comment:
what's the difference again between `support_partial_config` and specifying multiple `sharding_sources`?